### PR TITLE
:sparkles: Integrate AppCenter for Giskard

### LINKED ISSUE #1914: Integrate AppCenter with Giskard
We need them stats! Both analytics and crashing.

### PR DESCRIPTION
**Pull request by juanlaube**
Closes #1914 

This will enter in conflict with #1955, both changes are on the same class + project + packages.

### CODE REVIEW

File: Toggl.Giskard/Services/AnalyticsService.cs [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
     {
+        private const int maxAppCenterStringLength = 64;
+
         protected override void NativeTrackEvent(string eventName, Dictionary<string, string> parameters)
         {
+            AppCenterAnalytics.TrackEvent(eventName, trimLongParameters(parameters));
         }
 
         protected override void NativeTrackException(Exception exception)
         {
```
Comment (amulware) [inline Toggl.Giskard/Services/AnalyticsService.cs]: No exception tracking? :(

File: Toggl.Giskard/Services/AnalyticsService.cs [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
         }
+
+        private Dictionary<string, string> trimLongParameters(Dictionary<string, string> parameters)
+        {
+            var validParameters = new Dictionary<string, string>();
+            foreach (var (key, value) in parameters)
+            {
+                validParameters.Add(trimForAppCenter(key), trimForAppCenter(value));
```
Comment (amulware) [inline Toggl.Giskard/Services/AnalyticsService.cs]: This could throw if two trimmed keys are the same. Do we care? Will keys even ever be too long?

**Comment by amulware** [comment]
Apart from my comments, this looks pretty straightforward. :)

File: Toggl.Giskard/Services/AnalyticsService.cs (hunk 1 above)
Comment (juanlaube) [reply Toggl.Giskard/Services/AnalyticsService.cs]: Exception tracking is implemented in #1955.

File: Toggl.Giskard/Services/AnalyticsService.cs (hunk 2 above)
Comment (juanlaube) [reply Toggl.Giskard/Services/AnalyticsService.cs]: This is exactly how it is implemented in Daneel. I don't think it is a problem ¯\_(ツ)_/¯

File: Toggl.Giskard/Startup/SplashScreen.cs [hunk 3]
```diff
@@ -50,8 +50,9 @@ protected override void OnCreate(Android.OS.Bundle bundle)
 
             #if USE_ANALYTICS
             Microsoft.AppCenter.AppCenter.Start(
-                "{TOGGL_APP_CENTER_ID_DROID}",
-                typeof(Microsoft.AppCenter.Crashes.Crashes));
+            "{TOGGL_APP_CENTER_ID_DROID}",
```
Comment (amulware) [inline Toggl.Giskard/Startup/SplashScreen.cs]: Please fix indentation.

File: Toggl.Giskard/Services/AnalyticsService.cs (hunk 2 above)
Comment (willsb) [reply Toggl.Giskard/Services/AnalyticsService.cs]: > This is exactly how it is implemented in Daneel

Whenever this sentence is said, there is an opportunity for a refactor that encapsulates this logic in some shared portion of our codebase

ping @juanlaube

**Pull request merged**